spark/src/main/scala/org/apache/spark/sql/delta/DeltaTableIdentifier.scala

```diff
@@ -89,10 +89,10 @@ object DeltaTableIdentifier extends DeltaLogging {
     }
 
     spark.sessionState.conf.runSQLonFile &&
+      new Path(identifier.table).isAbsolute &&
```

Comment:
`isDeltaPath` is now called by `UpdateCatalog` on every potential update, so I have rearranged this a bit to ensure better short-circuiting.